### ISSUE TITLE
Erroe message when importing the solution

### ISSUE THREAD

**Issue by Ananass007** [joerodgers/sharepoint-app-registrations#1]
I'm getting an error message when importing the solution ''Missing dependencies'' new_sharedcommondataserviceforapps_9e64f

**Comment by alan-hu** [comment]
Beat me to it! I had the same issue. Love this concept and hope Joe can fix this.

**Comment by joerodgers** [comment]
I’ll take a look, haven’t seen this before.

**Comment by Ananass007** [comment]
Many thanks
Anass

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by alan-hu** [comment]
@joerodgers FYR, version 2.0.0.5 does not have this issue. It contains the required connection reference component. But v2.0.0.6 & v2.0.0.7 you probably left out the connection reference from the solution.

**Comment by Ananass007** [comment]
Many thanks, version 2.0.0.5 is imported correctly.
I have a question about the SPN permissions especially the sites.Full control.All, can’t we use a Sites.Selected instead?
Because if my understanding is correct, the full control will grant the SPN full control over all Sharepoint sites...


Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by alan-hu** [comment]
@Ananass007 , I figure the reason this SPN requires Sites.Fullcontrol.All is to inventory the other SPNs' permissions at each of the sites. This permission gives this SPN the ability to fire graph API calls to retrieve existing SPNs' permissions at a particular site. Refer to - https://learn.microsoft.com/en-us/graph/api/site-list-permissions?view=graph-rest-1.0&tabs=http

**Comment by joerodgers** [comment]
@Ananass007 is spot on.  It's also needed to pull a list of sites from the tenant.  Now that you have 2.0.05 installed, does it let you upgrade to 2.0.0.7?

**Comment by alan-hu** [comment]
@joerodgers No, in-place upgrade from v2.0.0.5 to v2.0.0.6 or v2.0.0.7 encounters the same missing dependencies error. See screen shot below:
<img width="378" alt="image" src="https://github.com/user-attachments/assets/805b3af6-d231-4ae2-a851-5e332620ee5f">

**Comment by Ananass007** [comment]
Hi Joe,
You solution works perfect in v2.0.0.5
It would be absolutely great if the flows can be extended to monitor even other delegated permissions, for example for exchange online.
Thank you very much.
Anass
________________________________

**Comment by Ananass007** [comment]
Hi Joe,
Importation was successful and running is smooth.
However, although I granted multip SPN accesss to sharepoint sites and assigned the site.selected permission on Entra to them, these sharepoint sites are not visible in the list. In fact the SharePoint column remains empty, but other columns for graph api and sharepoint api permission are working good.
Maybe some updates were implemented in the newer version that corrects this bug?
Because the real added value of the solution is the sharepoint sites list that are being accessed by SPNs
Many thanks,
Anass

________________________________

**Comment by Ananass007** [comment]
Hi Joe,

I managed to find the issue.
Assigning the SPN permission over a SharePoint site via the sharepoint site itself somehow did not work. (Via the /_layouts/15/appiny.aspx)
When doing it via Graph explorer via a Post, it did work and I am now able to see SharePoint site that the SPN has access to.

I would like to do a post on Linkedin about it but it would be useful if v2.0.0.6 or v2.0.0.7 are also working.

Many thanks for this useful tool.

Kind regards,
Anass

________________________________

**Comment by joerodgers** [comment]
@Ananass007, I fixed the package reference error in v2.0.0.7 and replaced the existing broken version.  Give it a shot.

**Comment by joerodgers** [comment]
@alan-hu, I fixed the package reference error in v2.0.0.7 and replaced the existing broken version. Give it a shot.

**Comment by Ananass007** [comment]
Thanks Joe,
I will check this asap
Kind regards
Anass

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________
